modules/imgcodecs/include/opencv2/imgcodecs.hpp

```diff
@@ -60,28 +60,34 @@ namespace cv
 //! @addtogroup imgcodecs
 //! @{
 
-enum { IMREAD_UNCHANGED  = -1, // 8bit, color or not
-       IMREAD_GRAYSCALE  = 0,  // 8bit, gray
-       IMREAD_COLOR      = 1,  // ?, color
-       IMREAD_ANYDEPTH   = 2,  // any depth, ?
-       IMREAD_ANYCOLOR   = 4,  // ?, any color
-       IMREAD_LOAD_GDAL  = 8   // Use gdal driver
+//! Imread flags
+enum ImreadModes {
+       IMREAD_UNCHANGED  = -1, //!< If set, return the loaded image as is (with alpha channel, otherwise it gets cropped)
+       IMREAD_GRAYSCALE  = 0,  //!< If set, always convert image to the single channel grayscale image
+       IMREAD_COLOR      = 1,  //!< If set, always convert image to the 3 channel BGR color image
+       IMREAD_ANYDEPTH   = 2,  //!< If set, return 16-bit/32-bit image when the input has the corresponding depth, otherwise convert it to 8-bit.
+       IMREAD_ANYCOLOR   = 4,  //!< If set, the image is read in any possible color format
+       IMREAD_LOAD_GDAL  = 8   //!< If set, use the gdal driver for loading the image
      };
 
-enum { IMWRITE_JPEG_QUALITY        = 1,
-       IMWRITE_JPEG_PROGRESSIVE    = 2,
-       IMWRITE_JPEG_OPTIMIZE       = 3,
-       IMWRITE_JPEG_RST_INTERVAL   = 4,
-       IMWRITE_JPEG_LUMA_QUALITY   = 5,
-       IMWRITE_JPEG_CHROMA_QUALITY = 6,
-       IMWRITE_PNG_COMPRESSION     = 16,
-       IMWRITE_PNG_STRATEGY        = 17,
-       IMWRITE_PNG_BILEVEL         = 18,
-       IMWRITE_PXM_BINARY          = 32,
-       IMWRITE_WEBP_QUALITY        = 64
+//! Imwrite flags
+enum ImwriteFlags {
+       IMWRITE_JPEG_QUALITY        = 1,  //!< For JPEG, it can be a quality from 0 to 100 (the higher is the better). Default value is 95.
+       IMWRITE_JPEG_PROGRESSIVE    = 2,  //!<
+       IMWRITE_JPEG_OPTIMIZE       = 3,  //!<
+       IMWRITE_JPEG_RST_INTERVAL   = 4,  //!<
+       IMWRITE_JPEG_LUMA_QUALITY   = 5,  //!<
+       IMWRITE_JPEG_CHROMA_QUALITY = 6,  //!<
+       IMWRITE_PNG_COMPRESSION     = 16, //!< For PNG, it can be the compression level from 0 to 9. A higher value means a smaller size and longer compression time. Default value is 3.
+       IMWRITE_PNG_STRATEGY        = 17, //!<
+       IMWRITE_PNG_BILEVEL         = 18, //!<
+       IMWRITE_PXM_BINARY          = 32, //!< For PPM, PGM, or PBM, it can be a binary format flag, 0 or 1. Default value is 1.
+       IMWRITE_WEBP_QUALITY        = 64  //!< For WEBP, it can be a quality from 1 to 100 (the higher is the better). By default (without any parameter) and for quality above 100 the lossless compression is used.
      };
 
-enum { IMWRITE_PNG_STRATEGY_DEFAULT      = 0,
+//! Imwrite PNG specific flags
+enum ImwritePNGFlags {
```

Comment:
These flags are nowhere to be discussed in current documentation nor do I have a clue what they are used for... Could you help out so that we can document this too. Where do we place these?